src/python/grpcio_tests/tests/unit/_contextvars_propagation_test.py

```diff
@@ -112,6 +114,46 @@ def test_propagation_to_auth_plugin(self):
                 response = stub(_REQUEST, wait_for_ready=True)
                 self.assertEqual(_REQUEST, response)
 
+    def test_concurrent_propagation(self):
+        _THREAD_COUNT = 32
+        _RPC_COUNT = 32
+
+        set_up_expected_context()
+        with _server() as port:
+            target = "localhost:{}".format(port)
+            local_credentials = grpc.local_channel_credentials()
+            test_call_credentials = TestCallCredentials()
+            call_credentials = grpc.metadata_call_credentials(
+                test_call_credentials, "test call credentials")
+            composite_credentials = grpc.composite_channel_credentials(
+                local_credentials, call_credentials)
+            start_event = threading.Event()
+
+            def _run_on_thread(exception_queue):
+                try:
+                    for _ in range(_THREAD_COUNT):
+                        with grpc.secure_channel(
+                                target, composite_credentials) as channel:
+                            start_event.wait()
```

Comment:
We may want a stronger synchronization here. The worker threads might be created by the testing thread, but the channel may not be created yet, so it might not be truly concurrent here.

E.g., we could have some counter, or list of events.

Reply:
Good point! Done.